src/vscode-dts/vscode.proposed.envCollectionWorkspace.d.ts

```diff
@@ -7,27 +7,32 @@ declare module 'vscode' {
 
 	// https://github.com/microsoft/vscode/issues/171173
 
-	export interface ExtensionContext {
+	// export interface ExtensionContext {
+	// 	/**
+	// 	 * Gets the extension's global environment variable collection for this workspace, enabling changes to be
+	// 	 * applied to terminal environment variables.
+	// 	 */
+	// 	readonly environmentVariableCollection: GlobalEnvironmentVariableCollection;
+	// }
+
+	export interface GlobalEnvironmentVariableCollection extends EnvironmentVariableCollection {
 		/**
-		 * Gets the extension's environment variable collection for this workspace, enabling changes
-		 * to be applied to terminal environment variables.
+		 * Gets scope-specific environment variable collection for the extension. This enables alterations to
+		 * terminal environment variables solely within the designated scope, and is applied in addition to (and
+		 * after) the global collection.
 		 *
-		 * @deprecated Use {@link getEnvironmentVariableCollection} instead.
-		 */
-		readonly environmentVariableCollection: EnvironmentVariableCollection;
-		/**
-		 * Gets the extension's environment variable collection for this scope, enabling changes
-		 * to be applied to terminal environment variables.
+		 * Each object obtained through this method is isolated and does not impact objects for other scopes,
+		 * including the global collection.
 		 *
 		 * @param scope The scope to which the environment variable collection applies to.
 		 */
-		getEnvironmentVariableCollection(scope?: EnvironmentVariableScope): EnvironmentVariableCollection;
+		getScoped(scope: EnvironmentVariableScope): EnvironmentVariableCollection;
 	}
 
 	export type EnvironmentVariableScope = {
 		/**
-		* Any specific workspace folder to get collection for. If unspecified, collection applicable to all workspace folders is returned.
-		*/
+		 * Any specific workspace folder to get collection for. If unspecified, collection applicable to all workspace folders is returned.
```

Comment:
I don't wish to make docstring of `getScoped` more verbose that it already is, if anything I think we could modify `EnvironmentVariableScope`:
```typescript
/**
 * Specifies the scope for which to retrieve the environment variable collection.
 * 
 * If a scope parameter is omitted, collection applicable to all relevant scopes for that parameter is returned.
 * For instance, if the 'workspaceFolder' parameter is not specified, the collection that applies across all
 * workspace folders will be returned.
 */
export type EnvironmentVariableScope = {
    /**
     * The specific workspace folder for which to retrieve the environment variable collection.
     */
    workspaceFolder?: WorkspaceFolder;
};
```

Reply:
>  Also should it throw if getting an undefined instead since it doesn't make sense to do, instead of silently failing and returning the global one?

I don't think so, if none of the parameters are specified, this will return the global collection by definition. We're making consuming of the API difficult if we throw in that case.

Reply:
> I don't wish to make docstring of getScoped more verbose than it already is, if anything I think we could modify EnvironmentVariableScope:

It's describing the behavior of `getScoped` though, `EnvironmentVariableScope` is just a set of options so describing what it does when everything is undefined doesn't make as much sense there. I think the `@param scope` line is ideal as it's describing what the argument does, not the method or object interface.

Reply:
I've added the description next to `@param scope` 👍